openquakeplatform/openquakeplatform/static/js/hrde_input_hazard_chart.js

```diff
@@ -0,0 +1,360 @@
+            //.attr("cx", 50)
+            .attr("cy", 0+(k*20))
+            .attr("cx", 460)
+            .attr("r", 3)
+            .style("fill", color)
+
+        $("."+selectedCurves[k]).css({'stroke':colors[k]});
+    }
+
+    for (i = 0 ; i < xAxis_n ; i++) {
+        var g = svg.append("g");
+        g.attr("class", "x axis")
+        g.attr("transform", "translate(0," + height + ")")
+        .call(xAxis[i]);
+        if (i == (xAxis_n - 1))
+            g.append("text")
+            .attr("x", width / 2)
+            .attr("y", 30)
+            .attr("dy", ".71em")
+            .attr("text-anchor", "middle")
+            .style("font-size","12px")
+            .text(xAxisLable);
+    }
+    svg.append("g")
+        .attr("class", "y axis")
+        .call(yAxis)
+        .append("text")
+        .attr("transform", "rotate(-90)")
+        .attr("y", -60)
+        .attr("x", -120)
+        .attr("dy", ".71em")
+        .style("font-size","12px")
+        .style("text-anchor", "middle")
+        .text(yAxisLable);
+
+    textTopTitle = svg.append("text")
+        .attr("x", 0)
+        .attr("y", -47)
+        .attr("dy", ".35em")
+        .style("font-weight", "bold")
+        .attr("font-size","14px")
+        .text(mappedValue1);
+
+    textTopTitle2 = svg.append("text")
+        .attr("x", 0)
+        .attr("y", -30)
+        .attr("dy", ".35em")
+        .style("font-weight", "bold")
+        .attr("font-size","14px")
+        .text(mappedValue2);
+
+    textTop = svg.append("text")
+        .attr("x", 0)
+        .attr("y", -15)
+        .attr("dy", ".35em")
+        .style("font-size","11px")
+        .text('');
+
+    $('#chartDialog').append('<div>* Magnitude Frequency Distributions</div>');
+    $('#chartDialog').append('<div id="downloadCurve"><font color="blue">Download Curve</font></div>');
+    $('#downloadCurve').on("hover", function(){
+        $(this).css("cursor", "pointer");
+    });
+
+    var h = $("#chartDialog").height();
+    h = h + 40;
+    $("#chartDialog").css({"height": h+"px"});
+
+    // Prep data for download to CSV
+    $('#downloadCurve').click(function(event) {
+        var csvHeader = selectedCurves;
+        var csvData = [];
+
+        //csvData = csvData.concat(csvHeader);
+        csvData.push("mfds");
+        csvData.push("binWidth");
+        csvData.push("minMag");
+        csvData.push("occurRate");
+        csvData.push("mags");
+        csvData = JSON.stringify(csvData);
+        var lineBreak = "lineBreak";
+        csvData = csvData.concat(lineBreak);
+        var quotationMark = '"';
+
+        csvData = csvData
+            .replace(/lineBreak/, '\r\n')
+            .replace(/\[/g, '')
+            .replace(/\]/g, '')
+            .replace(/","/g, ',')
+            .replace(/"/g, '');
+
+        for (var k in selectedCurves) {
+            curve_name = selectedCurves[k];
+            var curveValue = mfdsJsonObj[curve_name];
+
+            csvData += curve_name;
```

Comment:
Are we sure that curve_name don't includes single or double quote that must be escaped to fit in a valid CSV file ?